2020q2/linuxulator.md

```diff
@@ -9,20 +9,27 @@ Current focus is on applications that don't involve X11, mostly
 because they tend to be easier to test and debug, and the bug fixes
 are not application-specific.
 
-Example problems fixed include buggy madvise(2) handling, uname(2)
-returning wrong results for 32 bit apps, recvmsg(2) and accept(2) being
-broken in some circumstances, missing support for `SO_REUSEPORT`,
-`SO_SNDBUFFORCE`, `SO_RCVBUFFORCE`, and `SO_PROTOCOL`, default open files
-limit being way higher than Linux apps tend to assume, 
-
-There's ongoing work on debugging infrastructure and cleanups, such as
-logging warnings for unrecognized system call paramteres, or adding the
-`compat.linux.debug` sysctl to turn the warnings off.
+Example problems fixed include buggy madvise(2) handling, which could
+break applications linked against jemalloc; uname(2) returning wrong
+results for 32 bit apps, which caused problems for Steam; recvmsg(2)
+and accept(2) being broken in some circumstances, which was breaking
+Redis; and missing support for `SO_REUSEPORT`, `SO_SNDBUFFORCE`,
+`SO_RCVBUFFORCE`, and `SO_PROTOCOL`, which spammed the log files when
+running the Python regression test suite.  The default soft open files
+limit is now automatically adjusted to 1024, as several Linux apps
+iterate over all the file descriptors up to that limit instead of using
+closefrom(2).
+
+There's ongoing work on cleanups and the debugging framework for Linux
+compatibility, such as logging warnings for unrecognized system call
+parameters, or adding the `compat.linux.debug` sysctl to turn the warnings
+off.
 
 The Linux Test Project tests that are being run as part of the
 the [FreeBSD Continuous Integration infrastructure](https://ci.FreeBSD.org)
-has been upgraded to 20200605 snapshot.  This adds some more tests,
-and, predictably, some new failures.
+has been upgraded to 20200605 snapshot.  This raised the number
+of test cases from 3670 to 3749, and, predictably, also the number
+of failures, from 583 to 647.
```

Comment:
Mmm, fail rate went from 15.9% -> 17.3%

Not surprising that added tests are probably testing new functionality which we're least likely to support, but we might want to make a reference to that?

Reply:
I hadn't looked into that yet.  To me, LTP is useful when trying to test a specific piece of functionality, but there's no point in trying to make all the tests pass - LTP often tests useless stuff, such as specific errno values, instead of the actual functionality.  Thus the shift to more application-level test mechanisms.